### ISSUE TITLE
0.2.12: cannot load agent with --agent

### ISSUE THREAD

**Issue by jimlloyd** [cpacker/MemGPT#810]
**Describe the bug**
My agents are stored in the database. I can list agents with `memgpt list agents` and see the agents created so far:

```
+----------------------+-----------------------------+----------------+-------+-------------+---------------------------------------+
|         Name         |            Model            |    Persona     | Human | Data Source |              Create Time              |
+----------------------+-----------------------------+----------------+-------+-------------+---------------------------------------+
| KindheartedQuicksand |             None            | memgpt_starter | basic |             | 2024-01-13 11:12:18 AM UTC-08:00-0800 |
|   BountifulNugget    |             None            | memgpt_starter | basic |             | 2024-01-13 12:19:35 PM UTC-08:00-0800 |
|   PeacefulSquirrel   |             None            | memgpt_starter | basic |             | 2024-01-13 12:20:47 PM UTC-08:00-0800 |
|    QuirkyPenguin     | neuralhermes-2.5-mistral-7b | memgpt_starter | basic |             | 2024-01-13 11:47:33 AM UTC-08:00-0800 |
+----------------------+-----------------------------+----------------+-------+-------------+---------------------------------------+
```

I want to resume the session with QuirkyPenguin but `memgpt run --model neuralhermes-2.5-mistral-7b --agent QuirkyPenguin` fails with this error: 

```
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/cli/cli.py", line 489, in run
    memgpt_agent = Agent(agent_state, interface=interface)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/agent.py", line 181, in __init__
    if "system" not in agent_state.state:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable
```

**Please describe your setup**

- [ ] MemGPT version
  0.2.12

- [ ] How did you install memgpt?
  - poetry
- [ ] Describe your setup
  MacOS Sonoma
  Terminal in VS Code
 
**Screenshots**
N/A

**Additional context**
N/A

---

If you're not using OpenAI, please provide additional information on your local LLM setup:

**Local LLM details**
llama.cpp with neuralhermes-2.5-mistral-7b.Q8.gguf

If you are trying to run MemGPT with local LLMs, please provide the following information:

- [ ] The exact model you're trying to use (e.g. `dolphin-2.1-mistral-7b.Q6_K.gguf`)
- [ ] The local LLM backend you are using (web UI? LM Studio?)
- [ ] Your hardware for the local LLM backend (local computer? operating system? remote RunPod?)

**Comment by tombedor** [comment]
I think this is because of the state not being saved correctly in postgresql - check if the `state` column for your agent in the db is null

**Comment by sarahwooders** [comment]
I believed this was solved as part of #836 - basically the agent was being saved to the metadata store before the state had been initialized. I was able to re-load a saved agent but lmk if you're still having issues -- will close this for now.
